src/Compiler/Utilities/illib.fs

```diff
@@ -162,12 +162,17 @@ module internal PervasiveAutoOpens =
 
         static member RunImmediate(computation: Async<'T>, ?cancellationToken) =
             let cancellationToken = defaultArg cancellationToken Async.DefaultCancellationToken
+
             let ts = TaskCompletionSource<'T>()
+
             let task = ts.Task
 
             Async.StartWithContinuations(computation, (ts.SetResult), (ts.SetException), (fun _ -> ts.SetCanceled()), cancellationToken)
 
-            task.Result
+            try
```

Comment:
Why is this added?

Reply:
`NodeCode.RunImmediate` used to do this exception flattening, so probably to get it to 1:1 parity for it. At this moment I frankly don't remember if this was needed to get all the test passing.

Reply:
Not a big deal, but I noticed a dedicated "FlattenException" function in the PR, maybe it can be reused here as well?

Reply:
Yeah there is also such utility function in FSharp.Editor. There is room for some code clean-up. I'm just afraid if this PR stops passing CI I'll never get around to fixing it again :D

Reply:
As you wish :) If this starts causing mysterious problems, we can probably keep the current version. But I'd still try to just remove it to see if it still causes some test failures.

Reply:
I'll try removing it in a separate commit.

Reply:
Sure!

Reply:
OK, I did some investigation about this and refreshed my memory. Turns out, apart from tests, the only occurrence of this particular `Async.RunImmediate` we have in `main` happens here in `StackGuard.Guard`

https://github.com/dotnet/fsharp/blob/0115c93d599a9f9eaec2f974c66c1944231a4913/src/Compiler/Facilities/DiagnosticsLogger.fs#L907

The rest are `NodeCode.RunImmediate` and `NodeCode.RunImmediateWithoutCancellation`, which, unlike the `Async` version, do the unwraping of AggregateException.

Since I replaced all of that with `Async.RunImmediate` I had to add the unwrapping. For example this keeps green some tests that expect `OperationCancelledException` and not the `AggregateException` that the raw `TaskCompletionSource` emits.

If this ever causes any problem in the future it would be easy to restore the old behavior in that single spot.

TBH I'm not even sure if the current behavior of `StackGuard` wrapping `AggregateException` around the thrown exceptions is  the desired one. For sure changing it does not trip any tests we have.


To conclude, I think this is OK as is.

Reply:
Alright then, thanks for the explanation!